userspace/libsinsp/cri.hpp

```diff
@@ -656,85 +694,38 @@ inline std::string cri_interface<api>::get_container_image_id(const std::string
-	{
-		const auto pod_sandbox_id = root["sandboxID"].asString();
-		// Add the pod sandbox id as label to the container.
-		// This labels is needed by the filterchecks code to get the pod labels.
-		container.m_labels["io.kubernetes.sandbox.id"] = pod_sandbox_id;
-		typename api::PodSandboxStatusResponse resp_pod;
-		grpc::Status status_pod;
-		get_pod_sandbox_resp(pod_sandbox_id, resp_pod, status_pod);
-		if(status_pod.ok())
-		{
-			container.m_container_ip = ntohl(get_pod_sandbox_ip(resp_pod));
-			get_pod_info_cniresult(resp_pod, container.m_pod_cniresult);
-		}
-	}
-
-	return ret;
-}
-
 template<typename api>
 inline bool cri_interface<api>::parse(const libsinsp::cgroup_limits::cgroup_limits_key &key, sinsp_container_info &container)
 {
-	typename api::ContainerStatusResponse resp;
-	grpc::Status status = get_container_status(container.m_id, resp);
+	typename api::ContainerStatusResponse container_status_resp;
+	// status contains info around if API call suceeded and is not the container status property
+	grpc::Status status = get_container_status_resp(container.m_id, container_status_resp);
 
-	g_logger.format(sinsp_logger::SEV_DEBUG, "cri (%s): Status from ContainerStatus: (%s)", container.m_id.c_str(),
+	g_logger.format(sinsp_logger::SEV_DEBUG, "cri (%s): ContainerStatusResponse status error message: (%s)", container.m_id.c_str(),
 			status.error_message().c_str());
 
-	// If getting the container status fails then try to get the pod sandbox status.
+	// If container status failed try to get the pod sandbox status.
 	if(!status.ok())
 	{
-		typename api::PodSandboxStatusResponse resp;
-		grpc::Status status_pod;
-		get_pod_sandbox_resp(container.m_id, resp, status_pod);
-
-		if(status_pod.ok())
+		typename api::PodSandboxStatusResponse pod_sandbox_status_resp;
+		status = get_pod_sandbox_status_resp(container.m_id, pod_sandbox_status_resp);
+		if(status.ok())
 		{
+			/*
+			* We also want to ensure that the pod sandbox container stored in the container cache is
+			* fully filled out with available information as applicable.
+			* Most notably, the container's m_full_id and m_pod_sandbox_id will be the same, and the
+			* absence of container images can be attributed to the fact that they are not available for
+			* pod sandbox container processes.
+			*/
 			container.m_is_pod_sandbox = true;
-			// Fill the labels for the pod sanbox.
-			// Used to populate the k8s.pod.labels field.
-			for(const auto &pair : resp.status().labels())
-			{
-				if(pair.second.length() <= sinsp_container_info::m_container_label_max_length)
-				{
-					container.m_labels[pair.first] = pair.second;
-				}
-			}
+			const auto &resp_pod_sandbox_container = pod_sandbox_status_resp.status();
+			const auto &resp_pod_sandbox_container_info = pod_sandbox_status_resp.info();
+			const auto root_pod_sandbox = get_info_jvalue(resp_pod_sandbox_container_info);
+			parse_cri_base(resp_pod_sandbox_container, container);
```

Comment:
We previously didn't do a good job in handling pod sandbox containers and the labels (container or pod sandbox labels) were confusing in any aspect. Now each container regardless of whether it is a normal container or a pod sandbox container has `m_labels` and `m_pod_sandbox_labels`. 

Further improvements include adding all info that makes sense. Now pod sandbox container processes are also enriched the best way possible. Note that in the k8s filterchecks we do not need pod sandbox containers anymore to retrieve `m_pod_sandbox_labels`.